app/src/components/templateEditor/language/languageUtils.ts

```diff
@@ -0,0 +1,86 @@
+import { LRParser } from "@lezer/lr";
+
+/**
+ * Extracts all variables from a templated string.
+ *
+ * @param parser - The parser for the templating language.
+ *  The parser should be a language parser that emits Variable nodes.
+ * @param text - The text to extract variables from.
+ *
+ * @returns An array of variable names.
+ */
+export const extractVariables = (parser: LRParser, text: string) => {
```

Comment:
nit: we typically prefer "named" (object) arguments I would say for basically anything above 1 param